File: Programacao-Orientada-Objeto/Variaveis-e-Fluxo/Exercicio_7.java
import java.util.Scanner;

class Exercicio_7 {

    public static void main(String[] args){

        Scanner scan = new Scanner(System.in);
        int numeroX;
        
        System.out.print("Digite o número X: ");
        numeroX = scan.nextInt();

        while (numeroX == numeroX){

            if (numeroX % 2 == 0){

                numeroX = numeroX / 2;

            }
            else if (numeroX % 2 != 0){

                numeroX = 3 * numeroX + 1;
                
            }

            if (numeroX != 1){

                System.out.print(numeroX);

            }

            if (numeroX == 1){

                System.out.println(numeroX);

                break;

            }

            System.out.print(" -> ");

        }

    }

}

File: Programacao-Orientada-Objeto/Variaveis-e-Fluxo-CSharp/Exercicio-7.cs
using System;

class Exercicio_7 {

    static void Main(string[] args){

        int numeroX;
        
        Console.Write("Digite o número inicial: ");

        numeroX = Convert.ToInt32(Console.ReadLine());

        while (numeroX > -1){

            if (numeroX % 2 == 0) {

                numeroX = numeroX / 2;

            }

            else if (numeroX % 2 != 0) {

                numeroX = 3 * numeroX + 1;

            }

            if (numeroX != 1) {

                Console.Write(numeroX + " -> ");

            }

            else if (numeroX == 1){

                Console.WriteLine(numeroX);

                break;

            }

        }

    }

}